embedchain/cli.py

```diff
@@ -414,3 +437,99 @@ def deploy():
+    signal.signal(signal.SIGTERM, signal_handler)
+
+    # Check if 'api' and 'ui' directories exist
+    if os.path.exists("api") and os.path.exists("ui"):
+        pass
+    else:
+        # Step 1: Download the zip file
+        zip_url = "http://github.com/embedchain/ec-admin/archive/main.zip"
+        try:
+            response = requests.get(zip_url)
+            response.raise_for_status()
+            with tempfile.NamedTemporaryFile(delete=False) as tmp_file:
+                tmp_file.write(response.content)
+                zip_file_path = tmp_file.name
+            console.print("✅ [bold green]Downloaded zip file successfully.[/bold green]")
+        except requests.RequestException as e:
+            console.print(f"❌ [bold red]Failed to download zip file: {e}[/bold red]")
+            return
+
+        # Step 2: Extract the zip file
+        try:
+            with zipfile.ZipFile(zip_file_path, "r") as zip_ref:
+                # Get the name of the root directory inside the zip file
+                root_dir = Path(zip_ref.namelist()[0])
+                for member in zip_ref.infolist():
+                    # Build the path to extract the file to, skipping the root directory
+                    target_file = Path(member.filename).relative_to(root_dir)
+                    source_file = zip_ref.open(member, "r")
+                    if member.is_dir():
+                        # Create directory if it doesn't exist
+                        os.makedirs(target_file, exist_ok=True)
+                    else:
+                        with open(target_file, "wb") as file:
+                            # Write the file
+                            shutil.copyfileobj(source_file, file)
+                console.print("✅ [bold green]Extracted zip file successfully.[/bold green]")
+        except zipfile.BadZipFile:
+            console.print("❌ [bold red]Error in extracting zip file. The file might be corrupted.[/bold red]")
+            return
+
+        # Step 3: Install API requirements
+        try:
+            os.chdir("api")
+            subprocess.run(["pip", "install", "-r", "requirements.txt"], check=True)
+            os.chdir("..")
+            console.print("✅ [bold green]Installed API requirements successfully.[/bold green]")
+        except Exception as e:
+            console.print(f"❌ [bold red]Failed to install API requirements: {e}[/bold red]")
+            return
+
+    # Step 4: Start the API server
+    try:
+        os.chdir("api")
+        api_process = subprocess.Popen(
+            ["uvicorn", "main:app", "--reload", "--host", "127.0.0.1", "--port", "8000"], stdout=None, stderr=None
+        )
+        os.chdir("..")
+        console.print("✅ [bold green]API server started successfully.[/bold green]")
+    except Exception as e:
+        console.print(f"❌ [bold red]Failed to start the API server: {e}[/bold red]")
+        return
+
+    # Step 5: Install UI requirements and start the UI server
```

Comment:
@deshraj Step 5 and 6 are the same, why?

Reply:
Ah good catch. Will fix.